libsquoosh/src/worker_pool.ts

```diff
@@ -1,13 +1,15 @@
 import { Worker, parentPort } from 'worker_threads';
+// @ts-ignore
 import { TransformStream } from 'web-streams-polyfill';
+import type { JobMessage } from './index';
```

Comment:
I added the comment above but wanted to add here: wdyt about not importing something from `index` for the WorkerPool definition and implementation; to make it easier to reason about?

I think we can have a generic WorkerPool definition (essentially it dispatches a message (input) and returns a result (output), so it can be something like `class WorkerPool<I, O>`). Then we can use it in index as `WorkerPool<JobMessage, any>`. WDYT?